dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -1236,95 +1245,187 @@ def read_05evt_c_file(self):
+                    struct_terminal_subelem.tag == "comment":
+                pass  # nothing to do
+            else:
+                raise UnknownTerminalElement(struct_terminal_subelem.tag)
+
+        compo_name = struct_name + "." + name_terminal
+
+        # we store all this in an object
+        structure = Element(True, struct_name, len(self.list_elements))
+        if struct_name not in list_struct:
+            list_struct.append(struct_name)
+            self.list_elements.append(structure)
+            self.map_struct_name_struct_obj[struct_name] = structure
+        else:
+            structure = self.map_struct_name_struct_obj[struct_name]
+
+        # terminal name can be composed of a sub-structure (or several) and the actual name of the terminal
+        split_name = name_terminal.split('.')
+        struct_name1 = struct_name
+        structure1 = structure
+        if(len(split_name) > 1):  # terminal is composed of substructure:
+            list_of_structure = split_name[0:len(split_name) - 1]
+            for struct in list_of_structure:
+                struct_name1 = struct_name1 + "." + struct
+                if struct_name1 not in list_struct:
+                    s = Element(True, struct_name1, len(self.list_elements))
+                    list_struct.append(struct_name1)
+                    self.list_elements.append(s)
+                    self.map_struct_name_struct_obj[struct_name1] = s
+                    structure1.list_elements.append(s)
+                    structure1 = s
+                else:
+                    structure1 = self.map_struct_name_struct_obj[struct_name1]
+
+        terminal = Element(False, compo_name, len(self.list_elements))
+        self.list_elements.append(terminal)
+        structure1.list_elements.append(terminal)
+
+        # Find vars that are "flow"
+        if type_connector == "Flow" :
+            self.list_flow_vars.append(compo_name)
+
+    ##
+    # parse terminal etree element contained in elements etree element
+    # @param self : object pointer
+    # @param list_struct : structs list
+    # @param terminal_element : terminal etree element contained in elements etree element
+    # @return
+    def parse_terminal(self, list_struct, terminal_element):
+        number_of_terminal_names = 0
+        number_of_terminal_connector = 0
+        for elements_terminal_element in terminal_element:
+            if elements_terminal_element.tag == "name":
+                number_of_terminal_names += 1
+                if number_of_terminal_names >= 2:
+                    raise TerminalHasMoreThanOneName
+                name = elements_terminal_element.text
+            elif elements_terminal_element.tag == "connector":
+                number_of_terminal_connector += 1
+                if number_of_terminal_connector >= 2:
+                    raise TerminalHasMoreThanOneConnector
+                type_connector = elements_terminal_element.attrib.get('type', '')
```

Comment:
If connector XML element has no type attribute, type connector would be an empty string. But this modification is only necessary to make the unit tests pass so I'm not sure it's necessary.